core/src/main/kotlin/org/lflang/generator/ts/TSFileConfig.kt

```diff
@@ -54,13 +54,23 @@ class TSFileConfig(
         FileUtil.deleteDirectory(srcGenPath)
     }
 
+    fun setDocker(dockerEnabled: Boolean) {
+        docker = dockerEnabled
+    }
+
     override fun getCommand(): LFCommand {
-        return LFCommand.get(
-            "node",
-            listOf(srcPkgPath.relativize(executable).toString()),
-            true,
-            srcPkgPath
-        )
+        if (docker == true) {
```

Comment:
Why should the command be different depending on whether we use Docker or not?

Reply:
In this PR, we change to using the command in order to execute the program when we are running tests. In Docker mode, the command is no longer `node`; instead, the command that needs to be run is a command that executes the generated launcher script, which, in turn, executes `docker compose`.

Reply:
Why don't we just unify these use cases and execute a generated script in all cases? That would make most sense to me... In fact, this might already be what we're doing, except the test framework didn't catch up?

Reply:
In the test framework, that's what we were doing before we had this PR. But you're right, that's not what we've been doing in general. For most purposes we've been paying attention to the executables that the different targets say they produce; they say what executable they produce by overriding this method.

TypeScript does override the method, which caused a problem now that we changed the Docker tests to use the output of that method. One way to solve this is to essentially un-override by calling the superclass method when in Docker mode. It might be better to ignore the executables that the different targets say they produce and to assume that they all use a generated launcher script that is located in the same place, but it isn't obvious to me that that is better, and it wasn't important enough to me for me to think about it too hard. I'll be happy with whichever way we do it.

Reply:
I think all targets should either produce a binary or a script that is the entry point. That gives a uniform user experience and is the easiest to deal with in the rest of the toolchain.

Reply:
That makes sense, but the only downside is that it increases the chance that we will have issues related to running Bash scripts on Windows.

Reply:
I'd be OK with that. Most Windows users will use WSL if they are to use our tools from the command line.

Reply:
That is OK with me as well, but it is not consistent with the approach that we have been taking toward testing, so we might need to change some things to make that pass on Windows.

It also is not really in the same spirit as the kind of Windows support that we have claimed to have, since if the intended way to use LF on Windows is via WSL, we do not really need Windows support at all, and can probably just get by with Linux support. That is OK with me, too -- actually it is better than OK, because it means less work for everyone -- but it is a change that we might need people to be on board with.